OpenSpace/OpenSpace-Docs
# Releases/v0.20.0

This is the new version of the documentation that will be released for version 20.

To test:
* Code review `generatedocs.py`. Feedback is welcome since I don't code in python that much.
* Build the documentation and view it (see below for instructions). Feedback on anything is welcome, for example
    - Design
    - Structure
    - How well the current structure works for our documents 
    - Try it in dark and light mode (toggle this in the web inspector)

(A lot of the choices have been a joint effort, so some comments might not be able to act upon as we would have to discuss any changes)

Note that using the documentation locally is not going to show you the details in the search result list.

Some questions:
* Do we want to merge the RenderableOverview page before we have the Jenkins image tests?
* Do we want to merge the json files generated from the DocsWriter?
* What to do with the `About` section? It was decided to remove this section, but we haven't resolved where to put the content so I just added the section last for now.
* Do we want to remove empty pages? There are some placeholders

Building steps:
  1. In a shell, move to this directory.
  2. Run `python -m venv .venv` to create a python virtual environment.
  3. Run `.\.venv\Scripts\activate` to activate the python virtual environment.
  4. `pip install -r requirements.txt` (this is technically only necessary for the first time or when the dependencies change, but it is a good habit to run this every time after a checkout).
  5. Run `make html`.
